src/vivarium/framework/population.py

```diff
@@ -234,13 +230,18 @@ def update(self, population_update: Union[pd.DataFrame, pd.Series]):
                 )
             affected_columns = set(population_update.columns)
 
-        affected_columns = set(affected_columns).intersection(self._columns)
```

Comment:
I think removing this is going to have serious unintended consequences.

Reply:
It's not possible for affected columns to contain columns not in the view, since an error would have been thrown above. It was possible for this to happen if a Series was passed, but I've closed that loophole now.